calico/src/main/scala/calico/html/codecs/Codec.scala

```diff
@@ -89,10 +89,10 @@ private object Codec:
 
   val doubleAsString: Codec[Double, String] = new:
     def decode(domValue: String): Double =
-      domValue.toDouble // @TODO this can throw exception. How do we handle this?
+      domValue.toDoubleOption.getOrElse(null.asInstanceOf[Double])
     def encode(scalaValue: Double): String = scalaValue.toString
 
   val intAsString: Codec[Int, String] = new:
     def decode(domValue: String): Int =
-      domValue.toInt // @TODO this can throw exception. How do we handle this?
+      domValue.toIntOption.getOrElse(null.asInstanceOf[Int])
     def encode(scalaValue: Int): String = scalaValue.toString
```

Comment:
- `to{Double,Int}Option` is more efficient on error, because it does not create a `Throwable`.
- on JS everything is object, so it's safe to cast `null` to `Int` or `Double`
- now that `Codec` is internal private API, we can declare `decode` to be "unsafe" and return `null` in this situation. Actually that's what the Scaladoc says anyway
https://github.com/armanbilge/calico/blob/5c280cd4aac88268892234a234c8f39dab14229d/calico/src/main/scala/calico/html/codecs/Codec.scala#L37-L39
- do we use `Codec#decode` in our code anywhere? Couldn't find it 🤔 

WDYT?

Reply:
> * on JS everything is object, so it's safe to cast `null` to `Int` or `Double`

Hmm, Scala.js seems to be converting `null` to `0` 🤔 I wish it didn't do that 😂 

I guess for `Double` we can use `NaN`, but for `Int`, idk.

Reply:
We could also revert to the old behavior, which is `throw`ing. 🤷 

Or we can try your idea to use `Option` here. I was afraid it will require us to change the source generator so I felt lazy to do that 😅